pywbem/cim_obj.py

```diff
@@ -905,6 +881,113 @@ def __init__(self, classname, keybindings=None, host=None, namespace=None):
         self.host = host
         self.namespace = namespace
 
+    @property
+    def classname(self):
+        """
+        :term:`unicode string`: Name of the creation class of the referenced
+        instance.
+
+        Will not be `None`.
+
+        This attribute is settable. For details, see the description of the
```

Comment:
There should not be confusion re. which text applies to the constructor parameters snd which text applies to an attribute/property. In the description of ctor parms, I provide information about only that parm (possible types, preconditions, ...) it is written from a perspective of what matters when the user provides a value for the parameter (e.g. "must not be None".

For attributes/properties, we should use a consistent terminology. However, my interpretation of the two terms is that they denote two different things: An attribute is a feature of an object of a class. It may be public or private, and it may be implemented as a Python property. So to me, " property" denotes the implementation, while "attribute" denotes the general feature regardless of how it is implemented. I think I never validated this interpretation with the Python language docs, so that would be  a wortwhile exercise.

I chose "attribute" as a term because I did not want to point out the implementation.

The language I used is mainly for users to describe what they get when they read the attribute (e.g "will not be None"). In order to avoid duplication for settable properties, the description refers to the same-named ctro parm.

How die you mean your last question? Would we state that on ever, ctro parm that has a same-named settable attribute? I can live with that, but I don't like it because we already state that the ctor rules apply in the description of the settable attributes, and that is where it matters.